test-infra/src/main/java/com/datastax/oss/driver/api/testinfra/utils/ConditionChecker.java

```diff
@@ -25,12 +25,21 @@
 import java.util.concurrent.locks.ReentrantLock;
 import java.util.function.BooleanSupplier;
 
+/**
```

Comment:
we could change the internal API of `ConditionChecker` to use the `await()` underneath instead of our own solution. With such a solution there will be no need for deprecation, and the callers API will not change. But I am not sure we should hide the `await()` API this way.

Reply:
The goal is to get rid of `ConditionChecker` entirely. I would even have deleted it, but it's part of an "API" package and I want to be rigorous about backward compatibility (even though there's a good chance that nobody is using it outside of the driver).

Changing the internals is a good idea, I'll take a quick look at it.

Reply:
I'm not going to do it. It's doable, but not a trivial one-line change, we'd have to make sure the builder API still works. This is dead code now, so any time spent on it is a waste, even if it's just 10 minutes.